documentation/CAMARA-API-Design-Guide.md

```diff
@@ -0,0 +1,1165 @@
+
+#### 5.8.6. Security schemes
+
+[Security schemes](https://spec.openapis.org/oas/v3.0.3#security-scheme-object) express security in OpenAPI. 
+Security can be expressed for the API as a whole or for each endpoint.
+
+As specified in [Use of openIdConnect for securitySchemes](https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/CAMARA-API-access-and-user-consent.md#use-of-openidconnect-for-securityschemes), all CAMARA OpenAPI files must include the following scheme definition, with an adapted `openIdConnectUrl` in its components section. The schema definition is repeated in this document for illustration purposes, the correct format must be extracted from the link above.
+
+```yaml
+components:
+  securitySchemes:
+    openId:
+      type: openIdConnect
+      openIdConnectUrl: https://example.com/.well-known/openid-configuration
+```
+
+The key of the security scheme is arbitrary in OAS, but convention in CAMARA is to name it `openId`.
+
+## 6. Security
+
+### 6.1. Good practices to secure REST APIs
+
+The following points can serve as a checklist to design the security mechanism of the REST APIs.
+
+1. **Simple Management**. 
+   Securing only the APIs that need to be secure. Any time the more complex solution is made "unnecessarily", it is also likely to leave a hole. 
+2. **HTTPs must be always used**. 
+   TLS ensures the confidentiality of the transported data and that the server's hostname matches the server's SSL certificate.
+   - If HTTP 2 is used to improve performance, you can even send multiple requests over a single connection, this way you will avoid the complete overhead of TCP and SSL on later requests.
+
+3. **Using hash password**. 
+Passwords should never be sent in API bodies,
+   but if it is necessary it must be hashed to protect the system
+   (or minimize damage) even if it is compromised in some hacking attempts.
+   There are many hashing algorithms that can be really effective for password security,
+   for example, PBKDF2, bcrypt and scrypt algorithms.
+
+4. **Information must not be exposed in the URLs**
+Usernames, passwords, session tokens, and API keys should not appear in the URL, as this can be captured in web server logs, making them easily exploitable. For example, this URL (```https://api.domain.com/user-management/users/{id}/someAction?apiKey=abcd123456789```) exposes the API key. Therefore, never use this kind of security.
+
+5. **Authentication and authorization must be considered**
+   CAMARA uses the authentication and authorization protocols and flows as described in the [Camara Security and Interoperability Profile](https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/CAMARA-Security-Interoperability.md).
+
+6. **Add request time flags should be considered**. 

```

Comment:
This does not sound like a security consideration. Is there another section on general considerations?